_posts/2016-02-10-installing-server-environment-with-powershell.md

```diff
@@ -0,0 +1,199 @@
+
+## How stuff is installed without Chocolatey 
+If you are familiar with Windows Server installation you might notice that there are multiple different types of installation involved in the process. Windows features, executables, msi files and software specific plugins are all in the same process. This is the part where I wish that Windows PackageManagement (aka OneGet) or chocolatey can one day solve unifying installation. 
+
+#### Windows features (e.g. IIS)
+The first question you should be looking is "what are the windows features"? To be able to install the features you must first learn how to find them. Here is a oneliner that will give the available features to you. 
+
+```powershell
+Get-WindowsOptionalFeature -Online 
+```
+
+Great! Now we know how to find the windows features. Now we just need to change the oneliner a bit to be able to Install them. This is a oneliner for .NET:  
+
+```powershell
+Enable-WindowsOptionalFeature -Online -All -FeatureName NetFx4
+```
+
+All flag means that it will also install required dependencies if any. The .NET delivered among the operating system might not be latest but the feature needs to be enabled all the same. IIS will need it among some other features. Full list of my users can be found in our example config file at the bottom of the article.
+
+#### Executables 
+With windows feature we were able to install the .NET version that is delivered with the operating system. Although that is not most likely the newest one. Instead we need to fetch the newest one from [the internet](https://download.microsoft.com/download/E/4/1/E4173890-A24A-4936-9FC9-AF930FE3FA40/NDP461-KB3102436-x86-x64-AllOS-ENU.exe). With executables you never can be quite sure how they should be installed because there is no stricly unified commandline parameters. Here is an example how your executable installation might looklike. 
```

Comment:
that is not most likely the newest one -> that is, most likely, not the newest one.